postgres-types/src/time_03.rs

```diff
@@ -0,0 +1,109 @@
+use bytes::BytesMut;
+use postgres_protocol::types;
+use std::convert::TryFrom;
+use std::error::Error;
+use time_03::{Date, Duration, OffsetDateTime, PrimitiveDateTime, Time, UtcOffset};
+
+use crate::{FromSql, IsNull, ToSql, Type};
+
+#[rustfmt::skip]
+const fn base() -> PrimitiveDateTime {
+    PrimitiveDateTime::new(Date::__from_ordinal_date_unchecked(2000, 1), Time::MIDNIGHT)
```

Comment:
This does not seem like a public API?

Reply:
It's partially public. The problem is that we can `unwrap` in a `const` function, and calling that API seems less bad than bringing in a proc macro dependency.

Reply:
The `date!` macro might help here

https://docs.rs/time/0.3.2/time/macros/index.html

Reply:
Removed the `const` and switched to the public API.

@paolobarbolini I don't think that proc macro is worth its weight here.